SolidStateGroup/bullet-train-api
# feat: versioned feature states

Thanks for submitting a PR! Please check the boxes below:


## Changes

- Opt into feature state versioning via environment settings
- Create versioned segment overrides, environment feature state updates and multivariate options

What this does not fully cover, this should follow in an upcoming PR

- Change Requests workflow for versioning
- Creating/Deleting Multiviariate options

## How did you test this code?

- e2e tests that go through creating tests with feature versions enabled to check for regression

- Created and updated feature state remote config
![image](https://github.com/Flagsmith/flagsmith/assets/8608314/079876c2-8d37-4e67-9096-867682574d9f)

- Updated feature enabled state
![image](https://github.com/Flagsmith/flagsmith/assets/8608314/34e2956c-08d3-47fe-b6b2-57c70c0683a9)

- Created a segment override
![image](https://github.com/Flagsmith/flagsmith/assets/8608314/5fe519f2-5f9d-4b1a-ba97-4fca23ba80e1)

- Update segment override
Note: there is currently an API issue @matthewelwell is looking into.

- Update multivariate weightings
Note: this will probably change in the future, different feature versions could technically have totally different multivariate options, this currently only diffs the options that exist in the live feature.

![image](https://github.com/Flagsmith/flagsmith/assets/8608314/845988f8-3480-4f5f-bc4c-b266e9040b8a)
